gpu_stereo_image_proc_common/include/gpu_stereo_image_proc/msg_conversions.h

```diff
@@ -5,9 +5,26 @@
 #include <sensor_msgs/Image.h>
 #include <stereo_msgs/DisparityImage.h>
 
+class DisparityImageGenerator {
+  public: 
+    stereo_msgs::DisparityImagePtr getDisparity();
+    sensor_msgs::ImagePtr getDepth();
+    DisparityImageGenerator(
+      const sensor_msgs::ImageConstPtr &image, const cv::Mat_<int16_t> disparity16,
+      const image_geometry::StereoCameraModel &model, int min_disparity,
+      int max_disparity, int border, float downsample); 
+
+  private: 
+    stereo_msgs::DisparityImagePtr disparity;
+    sensor_msgs::ImagePtr depth;
+};
+
 stereo_msgs::DisparityImagePtr
```

Comment:
I **think** you can remove the standalone functions `disparityToDisparityImage` and `disparityImageToDepthImage`?

Reply:
If we're only using the VisionWorks part of the code (not libsgm or opencv parts), then we should remove it for sure. Otherwise, I'd have to replace calls to these functions in other parts of the code, and I'm not sure I can test if those end up working (I have libsgm skiplisted when building currently, I'm unsure if I can build it for testing if I do mess with the libsgm code).

Reply:
Oh yeah.   Forgot about that.   OK, leave it in place but add a \todo that we can remove them if/when the libsgm and opencv modules have been ported to use the class.    Long term, might as well port them because it's straightforward and then they can publish depth images.